js/user.js

```diff
@@ -1,24 +1,16 @@
-(function () {
-    let sidebar = document.getElementById("content-left");
-    let loadCommonCourses = null;
-
-    if (sidebar) {
-        let button = createButton("splus-user-courses-in-common-btn", "Courses in Common");
-        let img = createElement("img", [], { src: chrome.runtime.getURL("imgs/plus-icon.png"), width: 18, style: { verticalAlign: "middle", paddingLeft: "4px" } });
-        button.prepend(img);
-        button.querySelector("input").style.paddingLeft = "4px";
-        button.style.cursor = "pointer";
-        button.addEventListener("click", () => {
-            if (loadCommonCourses == null) {
-                loadCommonCourses = getCoursesInCommon(document.location.href.match(/\/(\d+)\//)[1]);
-            }
-            openModal("user-courses-in-common-modal", { loadCommonCourses: loadCommonCourses, userId: document.location.href.match(/\/(\d+)\//)[1] });
-        });
-
-        sidebar.appendChild(button);
-    }
+(async function () {
+    const page = document.getElementById("main-inner");
+    if (!page) return;
+    const userID = document.location.href.match(/\/(\d+)\//)[1];
+    const loadCommonCourses = getCoursesInCommon(userID);
+    const container = createElement("div", ["splus-modal-contents"], {}, [
+                        createElement("ul", ["setting-entry", "common-realm-list"], { id: "user-courses-in-common-list" }, [])
+                    ]);
+    const title = createElement("h4", [], {textContent: "Courses In Common"});
+    page.appendChild(title);
+    page.appendChild(container);
+    setCourseListModalContent({loadCommonCourses });
 })();

```

Comment:
I think for anonymous async functions like this one we like to have an error handler or some other way to make sure exceptions don't get swallowed. I believe grades.js has one if you want to model off of it. In lieu of a larger refactoring, simply logging the error should be fine.